core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -2422,16 +2423,16 @@ public void testRemoteLogSizeRetentionShouldFilterOutCopySegmentStartState()
         when(mockLog.logEndOffset()).thenReturn(2000L);
 
         // creating remote log metadata list:
-        // s1. One segment with "COPY_SEGMENT_STARTED" state to simulate the segment was failing on copying to remote storage.
+        // s1. One segment with "COPY_SEGMENT_STARTED" state to simulate the segment was failing on copying to remote storage (dangling).
         //     it should be ignored for both remote log size calculation, but get deleted in the 1st run.
         // s2. One segment with "DELETE_SEGMENT_FINISHED" state to simulate the remoteLogMetadataManager doesn't filter it out and returned.
         //     We should filter it out when calculating remote storage log size and deletion
-        // s3. One segment with "DELETE_SEGMENT_STARTED" state to simulate the segment was failing on deleting remote log.
-        //     We should count it in when calculating remote storage log size.
+        // s3. One segment with "DELETE_SEGMENT_STARTED" state to simulate the segment was failing on deleting remote log (dangling).
+        //     We should NOT count it when calculating remote storage log size and we should retry deletion.
         // s4. Another segment with "COPY_SEGMENT_STARTED" state to simulate the segment is copying to remote storage.
         //     The segment state will change to "COPY_SEGMENT_FINISHED" state before checking deletion.
         //     In the 1st run, this segment should be skipped when calculating remote storage size.
-        //     In the 2nd run, we should count it in when calculating remote storage log size.
+        //     In the 2nd run, we should count it in when calculating remote storage size.
         // s5. 11 segments with "COPY_SEGMENT_FINISHED" state. These are expected to be counted in when calculating remote storage log size
         //
         // Expected results (retention.size is 10240 (10 segments)):
```

Comment:
The comment in L2439/2440 should be updated.

Reply:
I just did a minor update, the rest looks correct to me. What's wrong?

Reply:
```
        // In the 1st run, the total remote storage size should be 1024 * 12 (s3, s5), so 2 segments (s1, s3) will be deleted
        // due to retention size breached. s1 will be deleted even though it is not included in size calculation. But it's fine.
        // The segment intended to be deleted will be deleted in the next run.
```
Since we won't count `DELETE_SEGMENT_STARTED` segment now, the total remote storage size should be 1024 * 11 , without s3. And s3 will be deleted is not because of retention size breach.

Reply:
Updated, let me know if it's ok now. Thanks.